src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java

```diff
@@ -76,23 +76,28 @@ public String deploy(AppDeploymentRequest request) {
 
 		logger.debug("Deploying app: {}", appId);
 
-		AppStatus status = status(appId);
-		if (!status.getState().equals(DeploymentState.unknown)) {
-			throw new IllegalStateException(String.format("App '%s' is already deployed", appId));
-		}
+		try {
+			AppStatus status = status(appId);
+			if (!status.getState().equals(DeploymentState.unknown)) {
+				throw new IllegalStateException(String.format("App '%s' is already deployed", appId));
+			}
 
-		int externalPort = 8080;
-		Map<String, String> parameters = request.getDefinition().getProperties();
-		if (parameters.containsKey(SERVER_PORT_KEY)) {
-			externalPort = Integer.valueOf(parameters.get(SERVER_PORT_KEY));
-		}
-		logger.debug("Creating service: {} on {}", appId, externalPort);
-		createService(appId, request, idMap, externalPort);
+			int externalPort = 8080;
+			Map<String, String> parameters = request.getDefinition().getProperties();
+			if (parameters.containsKey(SERVER_PORT_KEY)) {
+				externalPort = Integer.valueOf(parameters.get(SERVER_PORT_KEY));
+			}
+			logger.debug("Creating service: {} on {}", appId, externalPort);
+			createService(appId, request, idMap, externalPort);
 
-		logger.debug("Creating repl controller: {} on {}", appId, externalPort);
-		createReplicationController(appId, request, idMap, externalPort);
+			logger.debug("Creating repl controller: {} on {}", appId, externalPort);
+			createReplicationController(appId, request, idMap, externalPort);
 
-		return appId;
+			return appId;
+		} catch (Exception e) {
+			logger.error(e.getMessage(), e);
+			throw e;
```

Comment:
Actually, they would already be RuntimeExceptions, should change the catch to RuntimeException and the just re-throw it. Unless we would want to introduce a new DeploymentException in core.